packages/autocomplete-shared/src/decycle.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Decycles objects with circular references.
+ * This is used to print cyclic structures in development environment only.
+ */
+export function decycle(obj: any, seen: any[] = []) {
+  if (!__DEV__ || !obj || typeof obj !== 'object') {
```

Comment:
Making sure–did you check that there's no remainings of `decycle` in the production bundle?

Reply:
I wonder if we could write some tests on if the production bundle includes certain code, and throw an error?

Reply:
No, it's gone! I checked by tracking the string "[Circular]" in the production bundle since terser renames symbols. You can also see it from the Bundlesize check too (no change in bundle size).

Reply:
yes, I'd like that pattern in InstantSearch.js as well

Reply:
I created a [ticket](https://algolia.atlassian.net/browse/DX-2338) for this too. It could be an onboarding task if we don't get to do it before.